extensions/ql-vscode/src/common/events.ts

```diff
@@ -0,0 +1,10 @@
+import { Disposable } from '../pure/disposable-object';
+
+export interface AppEvent<T> {
+  (listener: (event: T) => any): Disposable;
```

Comment:
Can we make the return type of the `listener` either `void` or `unknown`? In theory it shouldn't return anything because we won't be using the result anyway, but I think it's good to make it explicit and explicitly require you to check the type.

Reply:
Yeah I can't see why not!